File: app/src/main/java/com/datarockets/mnchkn/fragments/charts/ChartsFragment.java
package com.datarockets.mnchkn.fragments.charts;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import com.datarockets.mnchkn.R;
import com.datarockets.mnchkn.adapters.PlayerChartListAdapter;
import com.datarockets.mnchkn.di.AppComponent;
import com.datarockets.mnchkn.fragments.BaseFragment;
import com.datarockets.mnchkn.fragments.charts.di.ChartsModule;
import com.datarockets.mnchkn.fragments.charts.di.DaggerChartsComponent;
import com.datarockets.mnchkn.models.Player;
import com.datarockets.mnchkn.utils.LogUtil;

import java.util.ArrayList;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import lecho.lib.hellocharts.model.Axis;
import lecho.lib.hellocharts.model.LineChartData;
import lecho.lib.hellocharts.view.LineChartView;

public class ChartsFragment extends BaseFragment implements ChartsView {

    public static final String TAG = LogUtil.makeLogTag(ChartsFragment.class);

    private static final String CHART_TYPE = "chart_type";

    @Inject ChartsPresenter presenter;
    View chartsView;
    LineChartData lineChartData;
    PlayerChartListAdapter lvPlayerListAdapter;
    @BindView(R.id.line_chart_view) LineChartView lineChartView;
    @BindView(R.id.lv_player_list) ListView lvPlayerList;

    public static ChartsFragment newInstance(int type) {
        Bundle args = new Bundle();
        args.putInt(CHART_TYPE, type);
        ChartsFragment fragment = new ChartsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle bundle) {
        super.onCreateView(inflater, container, bundle);
        chartsView = inflater.inflate(R.layout.fragment_charts, container, false);
        ButterKnife.bind(this, chartsView);
        return chartsView;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        lineChartData = presenter.loadChartData(getArguments().getInt(CHART_TYPE));
        lineChartData.setAxisXBottom(new Axis().setName(
                getResources().getString(R.string.text_steps)));
        lineChartData.setAxisYLeft(new Axis().setHasLines(true));
        lineChartView.setLineChartData(lineChartData);
        presenter.loadPlayersList(getArguments().getInt(CHART_TYPE) + 1);
    }


    @Override
    public void showPlayersList(ArrayList<Player> players) {
        lvPlayerListAdapter = new PlayerChartListAdapter(getActivity(), players,
                getArguments().getInt(CHART_TYPE));
        lvPlayerList.setAdapter(lvPlayerListAdapter);
    }


    @Override
    public void onDestroy() {
        super.onDestroy();
        presenter.onDestroy();
    }

    @Override
    protected void setUpComponent(AppComponent appComponent) {
        DaggerChartsComponent.builder()
                .appComponent(appComponent)
                .chartsModule(new ChartsModule(this))
                .build()
                .inject(this);
    }
}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/charts/ChartsFragment.kt
package com.datarockets.mnchkn.fragments.charts

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import butterknife.BindView
import butterknife.ButterKnife
import com.datarockets.mnchkn.R
import com.datarockets.mnchkn.adapters.PlayerChartListAdapter
import com.datarockets.mnchkn.di.AppComponent
import com.datarockets.mnchkn.fragments.BaseFragment
import com.datarockets.mnchkn.fragments.charts.di.ChartsModule
import com.datarockets.mnchkn.fragments.charts.di.DaggerChartsComponent
import com.datarockets.mnchkn.models.Player
import com.datarockets.mnchkn.utils.LogUtil
import lecho.lib.hellocharts.model.Axis
import lecho.lib.hellocharts.model.LineChartData
import lecho.lib.hellocharts.view.LineChartView
import java.util.*
import javax.inject.Inject

class ChartsFragment : BaseFragment(), ChartsView {

    @Inject lateinit var presenter: ChartsPresenter
    lateinit var chartsView: View
    lateinit var lineChartData: LineChartData
    lateinit var lvPlayerListAdapter: PlayerChartListAdapter
    @BindView(R.id.line_chart_view) lateinit var lineChartView: LineChartView
    @BindView(R.id.lv_player_list) lateinit var lvPlayerList: ListView

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, bundle: Bundle?): View? {
        super.onCreateView(inflater, container, bundle)
        chartsView = inflater!!.inflate(R.layout.fragment_charts, container, false)
        ButterKnife.bind(this, chartsView)
        return chartsView
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        lineChartData = presenter.loadChartData(arguments.getInt(CHART_TYPE))
        lineChartData.axisXBottom = Axis().setName(
                resources.getString(R.string.text_steps))
        lineChartData.axisYLeft = Axis().setHasLines(true)
        lineChartView.lineChartData = lineChartData
        presenter.loadPlayersList(arguments.getInt(CHART_TYPE) + 1)
    }


    override fun showPlayersList(players: ArrayList<Player>) {
        lvPlayerListAdapter = PlayerChartListAdapter(activity, players,
                arguments.getInt(CHART_TYPE))
        lvPlayerList.adapter = lvPlayerListAdapter
    }


    override fun onDestroy() {
        presenter.onDestroy()
        super.onDestroy()
    }

    override fun setUpComponent(appComponent: AppComponent) {
        DaggerChartsComponent.builder().appComponent(appComponent).chartsModule(ChartsModule(this)).build().inject(this)
    }

    companion object {

        val TAG = LogUtil.makeLogTag(ChartsFragment::class)

        private val CHART_TYPE = "chart_type"

        fun newInstance(type: Int): ChartsFragment {
            val args = Bundle()
            args.putInt(CHART_TYPE, type)
            val fragment = ChartsFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
